packages/text-annotator/src/SelectionHandler.ts

```diff
@@ -1,30 +1,50 @@
 import { Origin, type User } from '@annotorious/core';
 import { v4 as uuidv4 } from 'uuid';
 import type { TextAnnotatorState } from './state';
-import type { TextSelector, TextAnnotationTarget } from './model';
+import type { TextSelector, TextAnnotationTarget, TextQuoteSelector } from './model';
 import { trimRange } from './utils';
 
-export const rangeToSelector = (range: Range, container: HTMLElement, offsetReferenceSelector?: string): TextSelector => {
-  const rangeBefore = document.createRange();
+const rangeToQuoteSelector = (range: Range): TextQuoteSelector => {
+  const { startContainer, startOffset, endContainer, endOffset } = range;
+
+  const snippetLength = 10;
+
+  const rangePrefix = document.createRange();
+  rangePrefix.setStart(startContainer, Math.max(startOffset - snippetLength, 0));
+  rangePrefix.setEnd(startContainer, startOffset);
 
-  const offsetReference: HTMLElement = offsetReferenceSelector ? 
-    (range.startContainer.parentElement as HTMLElement).closest(offsetReferenceSelector) : container;
+  const rangeSuffix = document.createRange();
+  rangeSuffix.setStart(endContainer, endOffset);
+  rangeSuffix.setEnd(endContainer, Math.min(endOffset + snippetLength, endContainer.textContent.length));
+
+  return {
+    quote: range.toString(),
+    quotePrefix: rangePrefix.toString(),
+    quoteSuffix: rangeSuffix.toString()
+  }
+}
 
-  // A helper range from the start of the contentNode to the start of the selection
+export const rangeToSelector = (range: Range, container: HTMLElement, offsetReferenceSelector?: string): TextSelector => {
+  const offsetReference: HTMLElement = offsetReferenceSelector
+      ? (range.startContainer.parentElement as HTMLElement).closest(offsetReferenceSelector)
+      : container;
+
+  // Helper range from the start of the contentNode to the start of the selection
+  const rangeBefore = document.createRange();
   rangeBefore.setStart(offsetReference, 0);
   rangeBefore.setEnd(range.startContainer, range.startOffset);
 
-  const quote = range.toString();
+  const quoteSelector = rangeToQuoteSelector(range);
```

Comment:
BTW determining prefix/suffix could actually be simplified - the `rangeBefore` already contains all text before the quote. A `rangeAfter` could be constructed in a similar way.